src/App.jsx

```diff
@@ -14,33 +15,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [data, setData] = useState(goodsFromServer);
+  const [reversedStatus, setReversedStatus] = useState(false);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const customSort = (sortParam = 'alphabet') => {
+    return () => {
+      let sortedData = [...data];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+      switch (sortParam) {
+        case 'alphabet':
+          sortedData.sort((a, b) => a.localeCompare(b));
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+          if (reversedStatus === true) {
+            sortedData.reverse();
+          }
+
+          break;
+
+        case 'length':
+          sortedData.sort((a, b) => a.length - b.length);
+
+          if (reversedStatus === true) {
+            sortedData.reverse();
+          }
+
+          break;
+
+        case 'reverse':
+          sortedData.reverse();
+
+          if (reversedStatus === false) {
+            setReversedStatus(true);
+          } else {
+            setReversedStatus(false);
+          }
+
+          break;
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        default:
+          sortedData = goodsFromServer;
+          setReversedStatus(false);
+
+          break;
+      }
+
+      setData(sortedData);
+    };
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className="button is-info is-light"
+          onClick={customSort('alphabet')}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className="button is-success is-light"
+          onClick={customSort('length')}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={`button is-warning ${reversedStatus === false ? 'is-light' : ''}`}
+          onClick={customSort('reverse')}
+        >
+          Reverse
+        </button>
+
+        {reversedStatus === true && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={customSort('reset')}
+          >
+            Reset
+          </button>
+        )}
+      </div>
+
+      <ul>
+        {data.map(el => (
+          <li key={el} data-cy="Good">
+            {el}
+          </li>
```

Comment:
Good job on adding `key` to the list items during iteration. This is an important aspect of React that helps it identify which items have changed, are added, or are removed.